### ISSUE TITLE
Exclusion filter

### ISSUE THREAD

**Issue by slyguy65** [downthemall/downthemall#462]
Looking to add filters that exclude elements in a url. 
Not sure if already possible since there is no real reference guide to formatting.

P.S can we get a format/syntax guide so we can reference it when creating new filters.

**Comment by nmaier** [comment]
Yeah, it certainly would be good to have proper documentation for the filter syntax.

The short is:

1. You can use javascript regular expressions if you surround it with slashes, e.g. `/expr/`. Regular expressions can be quite powerful, but the syntax can be quite arcane and the concepts and usage can be steep to learn if you never came into contact with regular expressions before.
2. You can do "string matching" without slashes e.g. ".jpg" would match any url that has ".jpg" anywhere in it. 

As for the exclusion list. Yeah, I considered that in the past, and it was brought up in the past. However, I never implemented it, not because it's too difficult, but because the DownThemAll! user interface can be quite imposing as it is, and just dumping more and more "weird" features in it. I have yet to figure out how to do it in a non-confusing way.

I am open to suggestions of course, especially given that I am clearly not the best user-interface designer myself (I can only try my best).
